plugin/src/main/kotlin/com/automattic/android/CheckS3Version.kt

```diff
@@ -4,7 +4,8 @@ import java.net.HttpURLConnection
 import java.net.URL
 
 private const val SUCCESS_STATUS_CODE = 200
-private const val FAILURE_STATUS_CODE = 403
+private const val FORBIDDEN_STATUS_CODE = 403
+private const val FAILURE_STATUS_CODE = 404
```

Comment:
My initial reaction was "hang on, 404 is 'not found', not 'failure'". But ☝️ this really is signifying that if S3 sends us back a 404, we should interpret it as a failure. Moreover, if S3 starts sending a different HTTP code for a failure, the only change needed is replacing 404 for the new value, as opposed to renaming the constant 👍 👍 👍

Reply:
We can rename it to `not found` if you prefer. I used `FAILURE_STATUS_CODE = 403` exactly because of what you explained and instead of adding a new value for `404`, I've replaced the original one now that 404 is the expected failure code. Forbidden one is not necessary, but in case we revert the ACL settings, I don't want builds to fail, so 🤷